pandas/core/arrays/datetimes.py

```diff
@@ -2493,18 +2494,23 @@ def _maybe_localize_point(ts, is_none, is_not_none, freq, tz, ambiguous, nonexis
     return ts
 
 
-def generate_range(start=None, end=None, periods=None, offset=BDay()):
+def _generate_range(
+    start: Timestamp | None,
+    end: Timestamp | None,
+    periods: int | None,
+    offset: BaseOffset,
+):
```

Comment:
Confused why the default value for `offset` got dropped?

Reply:
bc this is never called without passing offset (outside tests, which this changes)